### ISSUE TITLE
Marks failing commands as having succeeded if they have too much output (SIGPIPE)

### ISSUE THREAD

**Issue by LukeShu** [nick-fields/retry#76]
**Describe the bug**
If a command has too much output, then even though the command might fail, the step gets marked as passing.

I measured "too much" as ≥1026 KiB, but I wouldn't be surprised if it's timing related and varies at any value much over 1024.

I've created a reproducer over at https://github.com/LukeShu/gha-test where each step `always()` runs and dumps an increasing amount of output to stdout and then exits with a failure. As seen in the screenshot below, everything <=1MiB correctly fails, but then everything >=1⅛MiB starts passing.

This can be replicated locally, as well:

(All you need to know about my `gha-test` reproducer is that `make -C ../gha-test bytes-1152` spits out 1152KiB of <=80-character lines on stdout, and then exits with code 1.)

```console
$ INPUT_CONTINUE_ON_ERROR=false INPUT_MAX_ATTEMPTS=1 INPUT_TIMEOUT_MINUTES=5 INPUT_COMMAND='make -C ../gha-test bytes-1152' node ./dist/index.js
…
9:  648 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
a:  729 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
b:  810 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
c:  891 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
d:  972 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
::debug::Code: null
::debug::Signal: SIGPIPE
Command completed after 1 attempt(s).

::set-output name=exit_code::0
```

I originally encountered this with v2.4.0, but the local-reproduction above is with the latest version (v2.7.0).

**Expected behavior**
I expected that if my tests fail, then my CI will fail.

**Screenshots**
![2022-06-14-164123_261x599_scrot](https://user-images.githubusercontent.com/90273/173701667-60c056af-d00d-4632-8d29-ca088d961d3c.png)

**Logs**
Enable [debug logging](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/enabling-debug-logging#enabling-step-debug-logging) then attach the [raw logs](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/using-workflow-run-logs#downloading-logs) (specifically the raw output of this action).

**Comment by tstackhouse** [comment]
Doing some research on this, it looks like this could be connected with the `maxBuffer` option of `child_process.exec()`, which defaults to `1024 * 1024` bytes, see the NodeJS docs: https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback

**Comment by nick-fields** [comment]
Thanks for this report. I just published v2.8.0 that will resolve this.  Thanks to your repo, I was able to reproduce the issue before the fix and then confirm it was fixed after by testing with upto 1000MiB of output.
